### PR TITLE
Replace jQuery document load event handler with javascript native version

### PR DESCRIPTION
**Pull request by DanielDervishi**
## Motivation and Context

- Replaces all `.ready()` jQuery method calls with an event listener on document handling the `DOMContentLoaded` event

## Your Changes

**Description**:
- Replaced all `.ready()` jQuery method calls with an event listener on document handling the `DOMContentLoaded` event or just executes the code that used to exist in the callback of the `.ready()` method calls (depending on the context)

**Type of change** (select all that apply):

- [x] Refactoring (internal change to codebase, without changing functionality)

## Testing

- Went through each individual context and tested that the behaviour was unchanged

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [ ] I have verified that the CI tests have passed. 
- [ ] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: app/views/tags/index.html.erb
Comment (DanielDervishi) [inline app/views/tags/index.html.erb:1]: Uses .ready() properly

File: app/views/groups/index.html.erb [hunk 1]
```diff
@@ -22,16 +22,16 @@
         }
       );
     });
-    $(function() {
-      $('#assignment-group-use').click(function(e) {
-        modalAssignmentGroupReUse.open();
-        $('#assignment-group-use-close').click(function(e) {
-          modalAssignmentGroupReUse.close();
+      document.addEventListener("DOMContentLoaded", function() {
```
Comment (david-yz-liu) [inline app/views/groups/index.html.erb:25]: Oh, I noticed that this function can be merged into the above (which is also a `DOMContentLoaded` callback)

File: app/views/tags/index.html.erb [hunk 2]
```diff
@@ -4,14 +4,17 @@
...
+        makeTagTable(document.getElementById('tags_table'), {course_id: <%= @current_course.id %>, assignment_id: <%= @assignment.id %>});
     });
-    $(function() {
-      $('#add-tag').click(function(e) {
-        modal_create_new.open();
-        e.preventDefault();
-      })
-    })
+    (function(){
+      const domContentLoadedCB = function() {
```
Comment (david-yz-liu) [inline app/views/tags/index.html.erb:10]: You can merge this callback into the one above.

File: app/views/tags/_boot.js.erb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <%= javascript_tag nonce: true do %>
-  $(document).ready(function() {
-    window.modal_create_new   = new ModalMarkus('#create_new_dialog');
-    window.modal_edit         = new ModalMarkus('#edit_tags_dialog');
-  });
+    document.addEventListener("DOMContentLoaded", function() {
```
Comment (david-yz-liu) [inline app/views/tags/_boot.js.erb:2]: Revert this indentation change

File: app/views/shared/_navigation_warning.js.erb [hunk 4]
```diff
@@ -15,11 +15,19 @@
...
     onbeforeunload_flag = true
-  })
+  });
 
-  $(function() {
-    $('input[type="submit"]').click(function () {
-      submit_clicked = true;
-    });
-  })
+  (function(){
```
Comment (david-yz-liu) [inline app/views/shared/_navigation_warning.js.erb:20]: Include a space before the `{`. Similarly below with spaces around the `else`

File: app/views/notes/_grouping.html.erb [hunk 5]
```diff
@@ -9,18 +9,26 @@
 <% end %>
 
 <%= javascript_tag nonce: true do %>
-  $(() => {
-    $('#assignment_id').on("change", (evt) => {
+  (function(){
```
Comment (david-yz-liu) [inline app/views/notes/_grouping.html.erb:12]: Include a space before the `{`, and below around `else`

**Review by david-yz-liu — COMMENTED** [review]
@DanielDervishi good work, I left a few different comments but overall I think this is quite close.

File: app/views/assignments/_boot.js.erb [hunk 6]
```diff
@@ -52,7 +52,7 @@
           <label class="required" for="'${input_id}'">${I18n.t('attributes.filename')}</label>
           <input type="text"
                  required="required"
-                 pattern="^[\-\._a-zA-Z0-9][\-\/\._a-zA-Z0-9\\s]*$"
+                 pattern="^[\\-\\._a-zA-Z0-9][\\-\\/\\._a-zA-Z0-9\\\\s]*$"
```
Comment (david-yz-liu) [inline app/views/assignments/_boot.js.erb:55]: This change isn't the one I was referring to. (In a regex, `-` at the beginning of a character class doesn't need to be escaped, nor does `.`)

Try adding a required file and then clicking the "delete" link to delete it, that's what caused me the error.

**Review by david-yz-liu — COMMENTED** [review]
@DanielDervishi great! I just left one comment about the "extra" assignment form bug I noticed.

**Review by david-yz-liu — APPROVED** [review]
Great work @DanielDervishi!

**Pull request merged**